### PR TITLE
GH-37096: [MATLAB] Add utility which makes valid MATLAB table variable names from an arbitrary list of strings

### LINKED ISSUE #37096: [MATLAB] Add utility which makes valid MATLAB `table` variable names from an arbitrary list of strings
### Describe the enhancement requested

To make it possible to safely convert Arrow `Schema` field names to corresponding MATLAB `table` variable names, it would be helpful to add a utility which can take an arbitrary list of strings and return a set of valid MATLAB `table` variable names, which are (1) unique, (2) non-empty, and (3) do not conflict with the "reserved" variable names "Properties", "VariableNames", "RowNames", and ":".

### Component(s)

MATLAB

### PR DESCRIPTION
**Pull request by sgilmore10**
### Rationale for this change

To make it possible to safely convert Arrow Schema field names to corresponding MATLAB `table` variable names, it would be helpful to add a utility which can take an arbitrary list of strings and return a set of valid MATLAB `table` variable names, which are (1) unique, (2) non-empty, and (3) do not conflict with the "reserved" variable names "Properties", "VariableNames", "RowNames", and ":". An additional restriction is that variable names must have 63 or less characters. 

### What changes are included in this PR?

1. Added a new function called `arrow.tabular.internal.makeValidVariableNames` that accepts an arbitrary list of strings and returns valid MATLAB `table` variable names.

```matlab
>> originalVarNames = ["", "Properties", ":", "ValidVar", "ValidVar"];
>> validVarNames = arrow.tabular.internal.makeValidVariableNames(originalVarNames)

validVarNames = 

  1×5 string array

    "Var1"    "Properties_1"    ":_1"    "ValidVar"    "ValidVar_1"
```

3. Added a new function called `arrow.tabular.internal.makeValidDimensionNames` that returns valid table dimension names with respect to a list of valid variable names. In MATLAB the default `table` dimension names are `"Row"` and `"Variables"`, but they must not conflict with any variables names. In other words, they must be unique with respect to the variable names. 

```matlab
>> validVarNames = ["Row" "Test" "Variables"];
>> validDimNames = arrow.tabular.internal.makeValidDimensionNames(validVarNames)

validDimNames = 

  1×2 string array

    "Row_1"    "Variables_1"
```

To summarize, MATLAB `table`s cannot have arbitrary variable names. For example, `"Properties"`, `"RowNames"`, `"VariableNames"`, and `":"` are all disallowed. Variable names must also be unique and must be between 1 and 63 characters in length. They also must be unique with respect to each other.

### Are these changes tested?

Yes. Added the following new test classes:

1. `tMakeValidVariableNames.m`
2. `tMakeValidDimensionNames.m`

### Are there any user-facing changes?

No.

### Future Directions

1. In a follow-up PR, we will integrate `makeValidVariableNames` and `makeValidDimensionNames` into the `table()` and `toMATLAB()` methods of `arrow.tabular.RecordBatch`.

### Notes

Thanks to @kevingurney for help writing the test cases!

* Closes: #37096

### CODE REVIEW

File: matlab/test/arrow/tabular/tMakeValidDimensionNames.m [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS,
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+% implied.  See the License for the specific language governing
+% permissions and limitations under the License.
+
+classdef tMakeValidDimensionNames < matlab.unittest.TestCase
+    
+    methods(TestClassSetup)
```
Comment (kevingurney) [inline matlab/test/arrow/tabular/tMakeValidDimensionNames.m:21]: This can be deleted.

File: matlab/test/arrow/tabular/tMakeValidDimensionNames.m [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        % Shared setup for the entire test class
+    end
+    
+    methods(TestMethodSetup)
```
Comment (kevingurney) [inline matlab/test/arrow/tabular/tMakeValidDimensionNames.m:25]: This can be deleted.

File: matlab/test/arrow/tabular/tMakeValidDimensionNames.m [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        % Setup for each test
+    end
+    
+    methods(Test)
+        % Test methods
```
Comment (kevingurney) [inline matlab/test/arrow/tabular/tMakeValidDimensionNames.m:30]: We can delete this comment.

File: matlab/test/arrow/tabular/tMakeValidVariableNames.m [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+            [actual, modified] = makeValidVariableNames(original);
+
+            testCase.verifyEqual(actual, expected);
+            testCase.verifyTrue(modified);
+        end
+
+        function VariableNameLengthMax(testCase)
+            % Verify strings whose character length exceeds 63
+            % are truncated to the max variable name length (63).
+
```
Comment (kevingurney) [inline matlab/test/arrow/tabular/tMakeValidVariableNames.m:227]: Can remove this empty line.

File: matlab/src/matlab/+arrow/+tabular/+internal/makeValidVariableNames.m [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+% distributed under the License is distributed on an "AS IS" BASIS,
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+% implied.  See the License for the specific language governing
+% permissions and limitations under the License.
+function [varnames, modified] = makeValidVariableNames(varnames)
+    arguments
+        varnames(1, :) string
+    end
+
+    reservedNames = ["Properties" "VariableNames" "RowNames", ":"];
```
Comment (kevingurney) [inline matlab/src/matlab/+arrow/+tabular/+internal/makeValidVariableNames.m:22]: It might be nice to use commas between all the elements here (currently there is only a comma after "RowNames").

**Review by kevingurney — CHANGES_REQUESTED** [review]
Thanks for working on this @sgilmore10! I just had a few minor comments.

**Review by kevingurney — APPROVED** [review]

**Comment by kevingurney** [comment]
+1
